File: backend/Services/TaskService.cs
using MongoDB.Driver;
using System.Collections.Generic;
using TaskModel = backend.Models.Task;

namespace backend.Services
{
    public class TaskService
    {
    private readonly IMongoCollection<TaskModel> _tasks;

        public TaskService(IConfiguration config)
        {
            var client = new MongoClient(config.GetConnectionString("TodoListDb"));
            var database = client.GetDatabase("todo_db");
            _tasks = database.GetCollection<TaskModel>("todo");
        }

        public List<TaskModel> Get() =>
        _tasks.Find(task => true).ToList();

        public TaskModel Get(string id) =>
            _tasks.Find<TaskModel>(task => task.Id.ToString() == id).FirstOrDefault();

        public TaskModel Create(TaskModel task)
        {
            _tasks.InsertOne(task);
            return task;
        }

        public void Update(string id, TaskModel t) =>
            _tasks.ReplaceOne(task => task.Id.ToString() == id, t);

        public void Remove(string id) =>
            _tasks.DeleteOne(task => task.Id.ToString() == id);
    }
}

File: frontend/src/services/task.service.js
import { httpService } from './http.service.js'


export const taskService = {
    query,
    getById,
    update,
    remove,
    add
}

async function query() {
    return httpService.get(`tasks`)
}

async function getById(id) {
    return httpService.get(`tasks/${id}`)
}

function remove(id) {
    return httpService.delete(`tasks/${id}`)
}

async function update(id, task) {
    console.log(task);
    
    return httpService.put(`tasks/${id}`, task)
}

function add(task) {
    return httpService.post('tasks', task)
}








